### ISSUE TITLE
camelcase Throwing Errors When Database Is In The Mix

### ISSUE THREAD

**Issue by ebebbington** [denoland/deno_lint#475]
Related project: https://github.com/drashland/deno-drash-realworld-example-app

I'm in the progress of a PR, and im currently making sure. it all abides by deno fmt and lint. But i'm getting camelcase errors:

![image](https://user-images.githubusercontent.com/47337480/97580360-18dbd080-19eb-11eb-98f4-8748d194e2a2.png)

And many more, i assume deno lint **shouldn't** be throwing these errors in this scenario judging by @bartlomieju's comment:

![image](https://user-images.githubusercontent.com/47337480/97580432-34df7200-19eb-11eb-9b91-0519f0e4030d.png)

Example code of what i'm working with:

```typescript
const sessionEntity: SessionModelEntity = {
    session_one: typeof session.session_one === "string"
      ? session.session_one
      : "",
    session_two: typeof session.session_two === "string"
      ? session.session_two
      : "",
    id: typeof session.id === "number" ? session.id : 0,
    user_id: typeof session.user_id === "number" ? session.user_id : 0,
};
return createSessionModel(sessionEntity);
```

**Comment by magurotuna** [comment]
Thanks for the report. I didn't take into account that such cases where object interfaces are defined externally. I'll try to fix it.

**Comment by ebebbington** [comment]
I am interested how you've took it into account currently - to me atleast, it seems impossible to allow camelcase for 'some db logic', whereas don't allow it for 'other stuff'

**Comment by magurotuna** [comment]
Currently, `camelcase` rule reports an error if a key in object literal is not in camelcase, regarding object literals. There's no other logic at the moment. 
You can see the description and some examples here: https://lint.deno.land/#camelcase, which should help you understand.

**Comment by ebebbington** [comment]
Ok cool thanks :)

**Comment by ebebbington** [comment]
Very good write up @magurotuna, i think that solution is the best of both worlds:

* I can still use snake_case for the 'entities'
* I don't need to ignore whole files or ignore the whole project (ie `deno lint --ignore-rule=camelcase`)
(By I i generally mean the user base)

But it means some refactoring will have to take place which is ok, nothing is perfect, but i like the solution described above 👍

**Comment by magurotuna** [comment]
@ebebbington Thanks for your feedback 😃  then I will fix both the hint message and the documentation, leaving the rule behavior as is.
